operator/cmd/mesh/manifest-apply.go

```diff
@@ -183,3 +202,41 @@ func InstallManifests(setOverlay []string, inFilenames []string, force bool, dry
 
 	return saveIOPToCluster(reconciler, iopStr)
 }
+
+// DetectIstioVersionDiff will show warning if istioctl version and control plane version are different
+// nolint: interfacer
+func DetectIstioVersionDiff(cmd *cobra.Command, tag string, kubeClient kube.ExtendedClient) error {
+	icps, err := kubeClient.GetIstioVersions(context.TODO(), controller.IstioNamespace)
+	if err != nil {
+		return err
+	}
+	if len(*icps) != 0 {
+		var icpTag string
+		for _, icp := range *icps {
+			tagVer, err := GetTagVersion(icp.Info.GitTag)
+			if err != nil {
+				return err
+			}
+			if tag != tagVer {
+				icpTag = tagVer
+			}
+		}
+		if tag != icpTag {
+			cmd.Printf("! istioctl version %s and the control plane version %s are different.\n"+
+				"! Proceed with caution because after installation the cluster may misbehave.\n\n", tag, icpTag)
```

Comment:
nit: extra new line at end

> after installation the cluster may misbehave.

This warning doesn't seem great. How is a user supposed to upgrade then? It sounds like we are saying if you upgrade your cluster is going to break

Reply:
> nit: extra new line at end

Removed.

> This warning doesn't seem great. How is a user supposed to upgrade then? It sounds like we are saying if you upgrade your cluster is going to break

I'm not really good at writing. Please suggest better wording so after noticing different versions user should be able to take necessary actions to avoid any future problems.